python/ray/serve/scripts.py

```diff
@@ -397,11 +403,40 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
             if property in deployment:
                 del deployment[property]
 
-    if output_path is not None:
-        if not output_path.endswith(".yaml"):
-            raise ValueError("FILE_PATH must end with '.yaml'.")
+    config_str = yaml.dump(
+        config, Dumper=ServeBuildDumper, default_flow_style=False, sort_keys=False
+    )
+    config_str += f"# Built using Ray v{ray.__version__}\n"
 
+    if output_path is not None:
         with open(output_path, "w") as f:
-            yaml.safe_dump(config, stream=f, default_flow_style=False, sort_keys=False)
+            f.write(config_str)
     else:
-        print(yaml.safe_dump(config, default_flow_style=False, sort_keys=False), end="")
+        print(config_str, end="")
+
+
+class ServeBuildDumper(yaml.SafeDumper):
+    """YAML dumper object with custom formatting for `serve build` command.
+
+    Reformat config to follow this spacing:
+    ---------------------------------------
+
+    import_path: example.path
+
+    runtime_env: {}
+
+    deployments:
+
+    - name: val1
+        ...
+
+    - name: val2
+        ...
+    """
+
+    def write_line_break(self, data=None):
+        # https://github.com/yaml/pyyaml/issues/127#issuecomment-525800484
+        super().write_line_break(data)
+
+        if len(self.indents) < 3:
+            super().write_line_break()
```

Comment:
what does this do?

Reply:
It adds line breaks between the deployment configurations and the top-level fields. It makes the YAML file look like this:

```yaml
import_path: ...

runtime_env: ...

deployments:

- name: deployment1
  num_replicas: 1
  max_concurrent_queries: 10
  ...

- name: deployment2
  num_replicas: 2
  max_concurrent_queries: 3
  ...
```

Without it, the file looks like this:

```yaml
import_path: ...
runtime_env: ...
deployments:
- name: deployment1
  num_replicas: 1
  max_concurrent_queries: 10
  ...
- name: deployment2
  num_replicas: 2
  max_concurrent_queries: 3
  ...
```

Reply:
I added a comment:

```Python
# Indents must be less than 3 to ensure that only the top 2 levels of
# the config file have line breaks between them. The top 2 levels include
# import_path, runtime_env, deployments, and all entries of deployments.
```